src/utils/array.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * Create array from results attributes
+ * @param data array of objects
+ * @returns array of attributes
+ */
+export const attributesFromDataArray = (data: any[]) => {
+  const propertiesArray: string[] = [];
```

Comment:
Wouldn't it make more sense to make this a set to begin with, so that you don't need to deduplicate after?